src/schema/README.md

```diff
@@ -87,371 +400,197 @@ reference__eeg:
+| `message` | Message for display to a user                  |
 
--   `number`: If `type` is `number`, then there MAY be any of the following
-    fields at the same level as `type`: `unit`,
-    `minimum` (inclusive minimum), `maximum` (inclusive maximum),
-    `exclusiveMinimum`, `exclusiveMaximum`.
-    Here is an example of a field with a minimum of 0, in units of
-    kilometers per second:
-    ```yaml
-    ExampleField:
-        name: ExampleField
-        description: |
-            The description of "ExampleField".
-        type: number
-        minimum: 0
-        # k stands for "kilo", m stands for "meter", "/" means "per", and
-        # s stands for "seconds"
-        unit: km/s
-    ```
+A level of `warning` corresponds to a rule in the specification that is RECOMMENDED, while a
+level of `error` corresponds to a rule that is REQUIRED.
 
--   `object`: If `type` is `object`, then there MAY be any of the following
-    fields at the same level as `type`: `additionalProperties`,
-    `properties`.
-    Objects are defined as sets of key-value pairs.
-    Keys MUST be strings, while values may have specific attributes,
-    which is what `additionalProperties` describes.
-    Here is an example of a field which MUST be an object,
-    with one optional field named "OptionalField" that MUST be a `string`
-    and any number of additional optional fields that MUST be `number`s.
-    ```yaml
-    ExampleField:
-        name: ExampleField
-        description: |
-            The description of "ExampleField".
-        type: object
-        properties:
-            OptionalField:
-                type: string
-        additionalProperties:
-            type: number
-    ```
+In some cases, an issue is contained next to a `level: required` or `level: recommended` as part
+of a larger rule. In these cases, the `level` field should be omitted from the issue to avoid
+duplication or conflict.
 
--   `boolean`: If `type` is `boolean`, then the field's value MUST be one
-    of: `true`, `false`. Here is an example:
-    ```yaml
-    ExampleField:
-        name: ExampleField
-        description: |
-            The description of "ExampleField".
-        type: boolean
-    ```
+### Filename construction rules
 
-Additionally, if multiple data types are possible for a given field
-(for example, if a field may be a `number` or an `array` of `number`s),
-then an `anyOf` field MUST be used, in which an array of data type
-descriptions MUST be defined.
-Here is an example of one such case:
-```yaml
-ExampleField:
-    name: ExampleField
-    description: |
-        The description of "ExampleField".
-    anyOf:
-        - type: number
-        - type: array
-            items:
-            type: number
-```
+A significant portion of BIDS is devoted to the naming of files, and almost all file names consist
+of entities, a suffix, an extension, and a data type. Exceptions will be noted below.
```

Comment:
This documents the current state.

I would like to rename `rules.datatypes` to `rules.filenames` and include `rules.associated_data` and `rules.top_level_files` inside this.